### PR TITLE
Add empty state for the breach list

### PR DESCRIPTION
**Pull request by Vinnl**
# References: 
Jira: MNTOR-1086
Figma: https://www.figma.com/file/gvyyqS9egsGjsH0LUDqT1m/Breach-Resolution-Flow-Design?node-id=654%3A16313&t=I1S7G3TnYG1IGh0z-4

# Description

This adds an empty state for the list of breaches. The acceptance criteria also mentioned telemetry, but I assume that that is depending on @rhelmer's work?

# Screenshot (if applicable)

image

Screen Shot 2023-02-09 at 15 05 49

image

Screen Shot 2023-02-09 at 15 05 37

# How to test

1. Open the breaches page for an email address with zero breaches - it should show the "No breaches found" message and hide the "Unresolved / Resolved" tabs.
2. Open the breaches page for an email address with more than zero unresolved breaches - it should just list those breaches.
3. Open the breaches page for an email address with more than zero resolved, and no unesolved breaches - it should show the "All breaches resolved" under the "Unresolved" tab, and list all breaches under the "Resolved" tab.
4. 1 and 3 above should show an "Add email address" button if the user has not reached their email address quota. Once they do, it should be hidden.

# Checklist (Definition of Done)
- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug. - Don't know how to - do we have unit tests for UI?
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met. - No telemetry, strings haven't been localised yet, and a couple of open questions inline.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/views/partials/breaches.js
Comment (Vinnl) [inline src/views/partials/breaches.js:93]: I just reused the `primary` class, but in the designs, it uses the Metropolis font and a font weight of 600. Should I update the `primary` class, or change it just for this instance?

File: src/client/css/partials/breaches.css
Comment (Vinnl) [inline src/client/css/partials/breaches.css:295]: These and the font settings for the `p`s above don't seem to have any effect. I expect that the (Metropolis) fonts aren't initialised for these weights? Was that a conscious decision?

File: src/views/partials/breaches.js
Comment (Vinnl) [inline src/views/partials/breaches.js:149]: Note that I'm just including the same string here multiple times - one for every of the user's email address. This is because I need to be able to toggle between the different strings on the client side, when the user selects a different email address, but of course becomes pretty cumbersome (or in some cases, even impossible) if we have to do this for every variable that changes on the client side. Do we have better solutions for this?

**Review by Vinnl — COMMENTED** [review]
Added some of my open questions inline. Other open question is whether I need to add telemetry already and if so, how?

File: locales/en/breaches.ftl
Comment (flodolo) [inline locales/en/breaches.ftl:48]:
```suggestion
breaches-all-resolved-cta-button = Add email address
```

It's nice to know from the ID where the string is used. Same for `breaches-none-cta`.

**Review by flodolo — APPROVED** [review]

File: src/views/partials/breaches.js
Comment (flozia) [reply src/views/partials/breaches.js:150]: Do you mean another way than directly putting them in the l10n strings or like the `<a></a>` tag for the string with ID `breach-checklist-pw-header` in `locales/en/breaches.ftl`?

File: src/views/partials/breaches.js
Comment (flozia) [reply src/views/partials/breaches.js:93]: We probably want to stay consistent with the buttons. I’d advocate for reusing the `primary` class for now and we can leave a comment for UX to see if they can adjust the design accordingly or otherwise change it respectively.

File: src/client/css/partials/breaches.css
Comment (flozia) [reply src/client/css/partials/breaches.css:295]: Only Metropolis Bold `700` is initialized in `src/client/css/fonts.css`. I guess that’s probably just due to the zero state using it for the first time?

File: src/client/css/partials/breaches.css
Comment (flozia) [inline src/client/css/partials/breaches.css:269]: Nit (non-blocking): Would be nice to have a comment on what state we are targeting here.

**Review by flozia — COMMENTED** [review]
This is working great!

I replied to your inline comments. The only thing I think should be addressed before landing this PR is adding the missing Metropolis weights and possible changes depending on what decision we land on regarding the multiple strings.

File: src/views/partials/breaches.js
Comment (Vinnl) [reply src/views/partials/breaches.js:150]: Ah yes, I guess you can include them directly like that `<a>`, but (for other use cases) I'd like to be able to set its attributes, potentially dynamically. In Fluent's React bindings, for example, you can pass an `elems` argument that looks somewhat like this:

```jsx
elems: {
  a: <a href={`https://some-dyamic.link/${userId}`} className="some-class"/>,
},
```

which results in `<a>` tags in the source string get replaced by the given element.

File: src/client/css/partials/breaches.css
Comment (Vinnl) [reply src/client/css/partials/breaches.css:296]: Ah gotcha. I added them here, of course at the cost of two additional resources to fetch: 97351dbf

File: locales/en/breaches.ftl
Comment (Vinnl) [reply locales/en/breaches.ftl]: Heh, I actually went back and forth on that :sweat_smile: Changed in 731b7a05.

File: src/client/css/fonts.css
Comment (toufali) [inline src/client/css/fonts.css:219]: Sorry for the lack of context RE fonts:
It was discussed that we try to keep Refresh fonts to a minimum for simplicity and page-load considerations. Inter should have a regular and bold weight, Metropolis (which coincidentally is no longer supported) should have a bold weight. Please try to conform to these options and let UX know during review. If there is UX concern we can add new fonts.

File: src/client/css/partials/breaches.css
Comment (toufali) [reply src/client/css/partials/breaches.css:296]: Ah, I was late to this party.  Yes it was a conscious decision – we agreed with UX that we would try to conform to a limited set of fonts to keep it simple and avoid extra client loads.

Same with heading and paragraph sizes – let's conform to the closest established variable and get UX sign-off.

For this case since it was already merged, I can chat with UX and post a follow-up PR with any adjustments.

File: src/views/partials/breaches.js
Comment (toufali) [reply src/views/partials/breaches.js:93]: Yes, as my other comments have trended – instead of creating too many new unique styles, default to using our established vars for consistency and get UX sign-off during review.

File: src/views/partials/breaches.js
Comment (toufali) [reply src/views/partials/breaches.js:149]: Yeah, I see the difficulty.  I posted a follow up and tagged you for review: https://github.com/mozilla/blurts-server/pull/2828

I think using fetch for the dynamic content as @flozia suggested would have worked too, but in this case we were already tracking the selected email via module state, so it wasn't much more of a stretch to inject dynamically.

The client/partials/breaches.js module needed touch up anyway (and will need more after [MNTOR-1160/add-new-email](https://github.com/mozilla/blurts-server/tree/MNTOR-1160/add-new-email) lands).  Using `<template>` was not required, but I opted for it instead of dealing with toggling between div `hidden` states.

File: src/client/css/fonts.css
Comment (Vinnl) [reply src/client/css/fonts.css:219]: Addressed in https://github.com/mozilla/blurts-server/pull/2825#discussion_r1104363695.

**Pull request merged**
